Loop/Plugins/PluginManager.swift

```diff
@@ -10,6 +10,13 @@ import Foundation
 import LoopKit
 import LoopKitUI
 
+
+public struct AvailableDevice {
```

Comment:
+1 I agree.  If the intent here is to anticipate onboarding flows driven by device plugins, should that really be part of the DIY merge?

Reply:
Yes, I agree with Darin's statement that "Available" device has been overloaded too much with Service things.  I think this should probably be renamed "AvailablePlugin", or have multiple types for each type of plugin. 

WRT Rick's comment, the onboarding has to be part of DIY merge, as Tidepool changes to remove the old settings interfaces and replace them with new ones that did not support unset values forced the implementation of onboarding for DIY, rather than update the new UIs to support unset values (which would eventually be throwaway code).

Reply:
I do think that the refactoring should be in a separate PR.  This PR is already too big and I'm on my 4th round of conflict resolution.